utils/context.go

```diff
@@ -0,0 +1,31 @@
+package utils
+
+import (
+	"context"
+
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/metadata"
+)
+
+const MetadataKey = "viam-metadata"
+
+func ContextWithMetadata(ctx context.Context) (context.Context, map[string][]string) {
```

Comment:
Do we need to return md here? Can it not be accessed via the ctx directly? https://pkg.go.dev/google.golang.org/grpc@v1.55.0/metadata? i.e. `FromIncomingContext`/`FromOutgoingContext`